### PR TITLE
Use send message protocol in cli

### PR DESCRIPTION
**Pull request by faassen**
# Description

Use the SendMessage protocol in the cli instead of `waspSays` and friends, by using `cliSendMessage` (and the lifted variant for commands, `cliSendMessageC`) instead. Instead of using `asWaspFailureMessage` everywhere there is a command error, move the full error handling into the command error handler itself.

Fixes # (issue)

## Type of change

Please select the option(s) that is more relevant.

- [x] Code cleanup
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Review by shayneczyzewski — CHANGES_REQUESTED** [review]
Hey there @faassen! I do like the idea of centralizing all the display functionality. There is one aspect of this change, however, I wanted to make you aware of. Previously, when we had things like:

```haskell
     CommandError $
        asWaspFailureMessage "Wasp command failed:"
          ++ ( "Couldn't find wasp project root - make sure"
                 ++ " you are running this command from a Wasp project."
             )
```

The `asWaspFailureMessage` was in change of styling the title of the message, and whatever was appended after followed that title like so:
Screen Shot 2022-03-07 at 7 48 40 AM

This change pulls out that `asWaspFailureMessage` into the top level printing, making the _entire message_ the title:
Screen Shot 2022-03-07 at 7 50 07 AM

If we move asWasp* into the top level printing, we probably want a way to distinguish between the title and body. The original intent of the title was to make it easier to see what was happening at a glance... in larger errors I personally think having them separated is nice.

I'll wait on a full review until we have time to chat on this more. 👍🏻 Thanks!

**Edit: crux is `asWaspFailureMessage` takes a `String` and returns a formatted `String`**
```
λ> putStrLn $ asWaspFailureMessage "header message: " ++ "lots \n of \n other stuff\n"


❌ --- [Error] header message:  ---------------------------------------------------

lots 
 of 
 other stuff
```

Which has the same precedence:
```
λ> putStrLn $ (asWaspFailureMessage "header message: ") ++ "lots \n of \n other stuff\n"


❌ --- [Error] header message:  ---------------------------------------------------

lots 
 of 
 other stuff
```

But different than what we are doing now:
```
λ> putStrLn $ asWaspFailureMessage ("header message: " ++ "lots \n of \n other stuff\n")


❌ --- [Error] header message: lots 
 of 
 other stuff
 ---------------------------
```

File: waspc/cli/Wasp/Cli/Command/Build.hs
Comment (Martinsos) [inline waspc/cli/Wasp/Cli/Command/Build.hs:47]: Does `alphaWarningMessage` already contain some styling? Might that be a problem? Because I see that before it was directly used with `waspSaysC`.

File: waspc/src/Wasp/Message.hs
Comment (Martinsos) [inline waspc/src/Wasp/Message.hs:12]: I would rename `Notification` to `Info`, as I feel like that it is used more commonly as a term for this kind of stuff. Most often log levels are info, warn and error.

**Review by Martinsos — COMMENTED** [review]
@faassen looking good to me! I did a very quick pass, looks good, left two minor comments. I will refrain from giving an explicit approval here -> instead I will let @shayneczyzewski do that since he was more recently than me working with this piece of code.

File: waspc/cli/Wasp/Cli/Command/Build.hs
Comment (faassen) [reply waspc/cli/Wasp/Cli/Command/Build.hs:47]: It was just a string, so now it's an actual warning. I think that makes sense as it's labeled "warning message".

File: waspc/src/Wasp/Message.hs
Comment (faassen) [reply waspc/src/Wasp/Message.hs:12]: Yeah, I was doubting about that. Note we our protocol already goes beyond info/warning/error. We have "start" already as a special form of notification, and "success"/"failure" in the end. I'll rename it to Info though.

**Comment by faassen** [comment]
@shayneczyzewski  thanks for pointing out the bug I introduced! It should be fixed on the latest commit. I've expanded the capability of Failure to take a title along with a message.

Eventually we may want to actually introduce a "phase" system where you can give a phase a title, and both success and failure using this title in their messaging. This is a small step in that direction.

File: waspc/cli/Wasp/Cli/Command/Compile.hs
Comment (shayneczyzewski) [reply waspc/cli/Wasp/Cli/Command/Compile.hs:68]: Original behavior:
Screen Shot 2022-03-07 at 1 55 41 PM

Behavior when combining them:
Screen Shot 2022-03-07 at 1 43 53 PM

You could in theory have two lines:
```haskell
      cliSendMessage $ Msg.Warning "Your project compiled with warnings:"
      cliSendMessage $ Msg.Info $ formatMessages warnings ++ "\n\n"
```
Screen Shot 2022-03-07 at 1 56 39 PM

Although they wouldn't have the same color, which is not ideal. Another option is to print each of the messages themselves:
```haskell
      cliSendMessage $ Msg.Warning "Your project compiled with warnings:"
      mapM_ (cliSendMessage . Msg.Warning) (zipWith (\i m -> show i ++ ") " ++ m) [1 :: Int ..] warnings)
```
Screen Shot 2022-03-07 at 1 58 05 PM

Or lastly, maybe we just remove the original "title" and just print them individually, like so:
```haskell
displayWarnings warnings = mapM_ (cliSendMessage . Msg.Warning) warnings
```
Screen Shot 2022-03-07 at 2 00 11 PM

I kinda like the last one...

Or some other way to get it looking like before? 🤷🏻‍♂️ Hopefully those options stir some ideas.

**Review by shayneczyzewski — APPROVED** [review]
Nice job cleaning all this up @faassen! I left one comment you can ideally address, and there is one Ormolu formatting error in `cli/Wasp/Cli/Command/Telemetry.hs` breaking CI you will want to fix before merging (trailing space at end, kinda hard to see in CI output), but otherwise have given my 👍🏻 Nice job

File: waspc/cli/Wasp/Cli/Command/Compile.hs
Comment (faassen) [reply waspc/cli/Wasp/Cli/Command/Compile.hs:68]: Thanks for catching this one, I overlooked that too. I will look into warnings.

Comment (faassen) [reply waspc/cli/Wasp/Cli/Command/Compile.hs:68]: There are only 2 warnings issued by the entire system.. So I will change the warning to use a heading too, just like failures.

**Pull request merged**
